addons/GDQuest_GDScript_formatter/plugin.gd

```diff
@@ -159,38 +194,52 @@ func update_shortcut() -> void:
 func _on_resource_saved(saved_resource: Resource) -> void:
 	if saved_resource is not GDScript:
 		return
-	if not get_editor_setting(SETTING_FORMAT_ON_SAVE):
-		return
 
-	var script := saved_resource as GDScript
+	var format_on_save := get_editor_setting(SETTING_FORMAT_ON_SAVE) as bool
+	var lint_on_save := get_editor_setting(SETTING_LINT_ON_SAVE) as bool
 
-	if not has_command(get_editor_setting(SETTING_FORMATTER_PATH)) or not is_instance_valid(script):
+	if not format_on_save and not lint_on_save:
 		return
 
-	var formatted_code := format_code(script, false)
-	if formatted_code.is_empty():
-		return
-
-	script.source_code = formatted_code
-	ResourceSaver.save(script)
-	script.reload()
-
-	var script_editor := EditorInterface.get_script_editor()
-	var open_script_editors := script_editor.get_open_script_editors()
-	var open_scripts := script_editor.get_open_scripts()
+	var script := saved_resource as GDScript
 
-	if not open_scripts.has(script):
+	if not has_command(get_editor_setting(SETTING_FORMATTER_PATH)) or not is_instance_valid(script):
 		return
 
-	if script_editor.get_current_script() == script:
-		reload_code_edit(script_editor.get_current_editor().get_base_editor(), formatted_code, true)
-	elif open_scripts.size() == open_script_editors.size():
-		for i: int in range(open_scripts.size()):
-			if open_scripts[i] == script:
-				reload_code_edit(open_script_editors[i].get_base_editor(), formatted_code, true)
-				return
-	else:
-		push_error("GDScript Formatter error: Unknown situation, can't reload code editor in Editor. Please report this issue.")
+	if format_on_save:
+		var formatted_code := format_code(script, false)
+		if formatted_code.is_empty():
+			return
+
+		script.source_code = formatted_code
+		ResourceSaver.save(script)
+		script.reload()
+
+		var script_editor := EditorInterface.get_script_editor()
+		var open_script_editors := script_editor.get_open_script_editors()
+		var open_scripts := script_editor.get_open_scripts()
+
+		if not open_scripts.has(script):
+			return
+
+		if script_editor.get_current_script() == script:
+			reload_code_edit(script_editor.get_current_editor().get_base_editor(), formatted_code, true)
+		elif open_scripts.size() == open_script_editors.size():
+			for i: int in range(open_scripts.size()):
+				if open_scripts[i] == script:
+					reload_code_edit(open_script_editors[i].get_base_editor(), formatted_code, true)
+					return
+		else:
+			push_error("GDScript Formatter error: Unknown situation, can't reload code editor in Editor. Please report this issue.")
+
+	if lint_on_save:
```

Comment:
A bit of weird behavior, there's some sort of race condition when linting on save where the highlighted lines will stay highlighted for a few seconds and then revert back to their original state.

When linting via the toolbar command, the lines stay highlighted.

Reply:
The compiler does its own warning check or form of linting, on its own timer that's often debounced for a couple of seconds by default, could that be linked to this? Or some other ticking process in the editor? I don't think the script editor really actually supports any plug-ins, we're hacking the CodeEdit node but otherwise the editor updates itself without really taking extensions into account.